[Ripple] Add max ripple radius property

codeman7 opened the pull request:
Add `maxRippleRadius` property to `MDCRippleView`. This allows clients to set the maximum size a ripple should expand if `rippleStyle` is set to `MDCRippleStyleUnbounded`. This property is part of the Ink component and in order to make transitions easier for clients from Ink to Ripple this property is necessary. Additionally this allows _icon button_ to be supported as a button with only an icon has a different ripple than a button without an icon or with an icon and text.

Closes #7338

codeman7 committed:
Make rippleRadius a property

codeman7 committed:
clang formatting

codeman7 committed:
Address feedback

yarneo committed:
Update MDCRippleLayer.m

codeman7 committed:
Update test

codeman7 committed:
clang

codeman7 committed:
Address feedback

Address feedback and add test where applicable. This adds a testing interface for MDCRippleLayer and renames a fake class to fit it's use case better. Also this updates the path when the rippleRadius is set.

codeman7 committed:
Clangy clang

codeman7 committed:
Merge branch 'ripple-refactor' into ripple-radius

codeman7 committed:
Initial commit

This includes the new property and test for that new property

material-automation commented:
bazel detected changes to the following targets:

```
//components/AppBar:snapshot_tests
//components/AppBar:snapshot_tests_test_bundle
//components/AppBar:unit_tests
//components/AppBar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/AppBar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/AppBar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/AppBar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/AppBar:unit_tests_IPHONE_X_IN_11_0
//components/AppBar:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/Banner:snapshot_tests
//components/Banner:snapshot_tests_test_bundle
//components/BottomAppBar:snapshot_tests
//components/BottomAppBar:snapshot_tests_test_bundle
//components/BottomAppBar:unit_tests
//components/BottomAppBar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/BottomAppBar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/BottomAppBar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/BottomAppBar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/BottomAppBar:unit_tests_IPHONE_X_IN_11_0
//components/BottomAppBar:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/ButtonBar:snapshot_tests
//components/ButtonBar:snapshot_tests_test_bundle
//components/ButtonBar:unit_tests
//components/ButtonBar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/ButtonBar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/ButtonBar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/ButtonBar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/ButtonBar:unit_tests_IPHONE_X_IN_11_0
//components/ButtonBar:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/Buttons:snapshot_tests
//components/Buttons:snapshot_tests_test_bundle
//components/Buttons:unit_tests
//components/Buttons:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Buttons:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Buttons:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Buttons:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Buttons:unit_tests_IPHONE_X_IN_11_0
//components/Buttons:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/Cards:snapshot_tests
//components/Cards:snapshot_tests_test_bundle
//components/Cards:unit_tests
//components/Cards:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Cards:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Cards:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Cards:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Cards:unit_tests_IPHONE_X_IN_11_0
//components/Cards:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/Chips:snapshot_tests
//components/Chips:snapshot_tests_test_bundle
//components/Chips:unit_tests
//components/Chips:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Chips:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Chips:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Chips:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Chips:unit_tests_IPHONE_X_IN_11_0
//components/Chips:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/Dialogs:snapshot_tests
//components/Dialogs:snapshot_tests_test_bundle
//components/Dialogs:unit_tests
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Dialogs:unit_tests_IPHONE_X_IN_11_0
//components/Dialogs:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/NavigationBar:snapshot_tests
//components/NavigationBar:snapshot_tests_test_bundle
//components/NavigationBar:unit_tests
//components/NavigationBar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/NavigationBar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/NavigationBar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/NavigationBar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/NavigationBar:unit_tests_IPHONE_X_IN_11_0
//components/NavigationBar:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/Ripple:snapshot_tests
//components/Ripple:snapshot_tests_test_bundle
//components/Ripple:unit_tests
//components/Ripple:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Ripple:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Ripple:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Ripple:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Ripple:unit_tests_IPHONE_X_IN_11_0
//components/Ripple:unit_tests_IPHONE_X_IN_11_0_test_bundle
//components/ShadowElevations:snapshot_tests
//components/ShadowElevations:snapshot_tests_test_bundle
//components/Snackbar:snapshot_tests
//components/Snackbar:snapshot_tests_test_bundle
//components/Snackbar:unit_tests
//components/Snackbar:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Snackbar:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Snackbar:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Snackbar:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Snackbar:unit_tests_IPHONE_X_IN_11_0
//components/Snackbar:unit_tests_IPHONE_X_IN_11_0_test_bundle
```

<!-- identifier: affected-targets -->

material-automation commented:
## API diff detected the following changes

### Ripple

#### MDCRippleView

*new* property: `maximumRadius` in `MDCRippleView`

<!-- identifier: apidiff -->

codeman7 committed:
Add documenation around tests

codeman7 committed:
Rename property

Rename property to use maximum instead of max and remove ripple since this is on the ripple object

codeman7 committed:
Update docs around test

jverkoey commented on components/Ripple/src/private/MDCRippleLayer.h:
```diff
@@ -45,6 +45,11 @@ typedef void (^MDCRippleCompletionBlock)(void);
+/**
+ The radius the ripple expands to when activated.
+ */
+@property(nonatomic, assign) CGFloat rippleRadius;
```
Should this be maxRippleRadius to match the other api?

jverkoey replied:
Put another way:  unless there is a technical reason to do otherwise I would expect both of the radius APIs to have identical names.

yarneo replied:
I feel similarly, unless you think the old name is incorrect or this new naming is significantly better.

codeman7 replied:
This API is on `MDCRippleLayer` which is coming from #7350, this PR also adds an API to `MDCRipppleView` and I will update that. I originally had it as `maxRippleRadius` but then just thought the name was a minor improvement. It seems this could cause more headache during migration so I'll go back to `maxRippleRadius`.

jverkoey replied:
@codeman7 and I discussed offline that consistency is important here. @codeman7 feels that `maximumRadius` is the most preferred API here, and I agree. The migration cost for clients will be an additional find-and-replace, but I consider this a low cost in relation to the value of aligning the naming on Objective-C naming conventions (e.g. `minimumFontSize` in UITextField).

The action item here then is to rename both APIs to `maximumRadius`.

yarneo replied:
SGTM, thanks!

yarneo commented on components/Ripple/src/MDCRippleView.m:
```diff
@@ -155,6 +155,9 @@ - (void)beginRippleTouchDownAtPoint:(CGPoint)point
   rippleLayer.rippleLayerDelegate = self;
   rippleLayer.fillColor = self.rippleColor.CGColor;
   rippleLayer.frame = self.bounds;
+  if (self.rippleStyle == MDCRippleStyleUnbounded) {
```
is this the only change needed? I recall in InkView it being much more elaborate. If it is, this is really great!

codeman7 replied:
`MDCInkView` holds a reference to the inkLayer and uses that within _legacy ink_, instead of creating a new layer every time. Additionally since it holds a reference to the layer it updates the `maxRippleRadius` if the style changes. Since Ripple creates a new layer every time this isn't needed.

yarneo replied:
Awesome! thanks for checking

codeman7 committed:
Remove setter

Since startRippleAtPoint calls setPathFromRadii we do not need the setter. Additionally if the setter is there and a rippleLayer is already active, in a selected state, this could cause unexpected behavior.

codeman7 committed:
Rename property

This renames the property based on feedback. Additionally this updates the function name to match the style guide

codeman7 committed:
Clang

codeman7 committed:
Merge branch 'ripple-refactor' into ripple-radius

codeman7 committed:
Remove testing interfaces

codeman7 committed:
Merge branch 'ripple-refactor' into ripple-radius

codeman7 committed:
Rename property to maxRippleRadius

codeman7 committed:
Rename maxRippleRadius to maximumRadius

codeman7 committed:
Merge branch 'ripple-refactor' into ripple-radius

codeman7 committed:
Update naming

codeman7 committed:
Merge branch 'develop' into ripple-radius
